android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginActivity.kt

```diff
@@ -56,7 +56,7 @@ class LoginActivity : BaseMultiLanguageActivity() {
       ) {
         loginService.navigateToPinLogin(false)
       } else if (accountAuthenticator.hasActiveSession() && !isPinEnabled) {
-        loginService.navigateToHome()
+        accountAuthenticator.localLogout()
```

Comment:
What would happen here is if the application is not using PIN login and there is an active session (active access token) the user will be forced to log out. Is this the desired use case?

Reply:
User will be required to login each time they use the application even if there is already an existing active session. I do not think this is correct.

Reply:
The changes affect the workflow when the user leaves the app (or the app is closed) and has no effect when the user goes back to the app if it was in the background. 
- Currently, if there's an  active session and the PIN is set, they'll always be taken to the PIN Login screen
- The changes introduce being directed to the Login screen if there's an active session and a PIN isn't enabled/set
This is based on the check `"Was PIN login setup = No"` workflow as detailed in this flow chart  

![image](https://user-images.githubusercontent.com/16164649/207549295-762ff10b-f8a1-4a53-ae13-2c5feb70c25c.png)

Reply:
This is a good question, I don't know the answer. @f-odhiambo correct if wrong, but I think the entire diagram above is describing what applies when there is not an active session. I assumed when there is an active session, no login required, but I'm not sure of that.

Reply:
@pld I think making the user log in every time they open the app they need to authenticate. The reason for this to make sure that the data is not accessed by un-authorised users. 

I think we don't need to care much about the sessions in the authentication flow becuase the auth token renewal process will handle that when the app is trying to sync data to the server.  Might be worng about this.